include/libtorrent/aux_/session_impl.hpp

```diff
@@ -563,6 +563,14 @@ namespace aux {
 			std::uint16_t ssl_listen_port() const override;
 			std::uint16_t ssl_listen_port(listen_socket_t* sock) const;
 
+			void for_each_listen_socket(std::function<void(aux::session_listen_socket*)> f) override
```

Comment:
since this is in the header already, perhaps it would make sense to make it a template and avoid type-erasure on the function object and allow inlining..